src/realmModels/Observation.js

```diff
@@ -371,27 +378,45 @@ class Observation extends Realm.Object {
       fields: { id: true }
     };
 
-    const response = await createObservation( uploadParams, options );
+    let response;
+
+    const wasPreviouslySynced = Observation.wasPreviouslySyncedObservation( realm, obs );
+
+    if ( wasPreviouslySynced ) {
+      response = await updateObservation( {
+        id: newObs.uuid,
+        ignore_photos: true,
+        observation: { ...newObs },
+        fields: { id: true }
+      }, options );
+    } else {
+      response = await createObservation( uploadParams, options );
+    }
+
     await Observation.markRecordUploaded( obs.uuid, "Observation", response, realm );
     const { id } = response.results[0];
-    if ( obs?.observationPhotos?.length > 0 ) {
+
+    // skip updating evidence for already synced observations
+    // since these .update API calls aren't yet available in v2
+
+    if ( obs?.observationPhotos?.length > 0 && !wasPreviouslySynced ) {
```

Comment:
Did you find this was necessary? I'm pretty sure `POST /v2/observation_photos` behaves as an upsert. `POST www.inaturalist.org/observation_photos` does, specifically to accommodate the mobile apps.

Reply:
I don't think this is acting exactly as expected. I'm getting an undefined response back from `observation_photos` which have already been uploaded, which makes me think they're not upserting & also means we can't update the synced date on the client side.

For now, I decided to filter by evidence that hasn't been synced yet, so this can add new observation photos to an observation, but it can't update or delete observation photos. I think this is fine for now, since we also don't have a way to update observation photos in the client (like moving position) and we don't yet have an API v2 endpoint or working functionality in the client for deleting observation photos.